src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey
+<T>(items: T[], key: keyof T): GroupsMap<T> {
+  // i know it is not good, but what type it will be?
+  const result: { [key: string]: T[keyof T] } = {};
+
```

Comment:
You lready have this type - use it
```suggestion
  const result: GroupsMap<T> = {};

```

Reply:
is not fixed

Reply:
@Deniskakaka Does it not work if you use `GroupsMap<T>` here?